File: service/BoxService.cs
using infrastructure;
using infrastructure.Models;

namespace service;

public class BoxService
{
    private readonly Repository _repository;

    public BoxService(Repository repository)
    {
        _repository = repository;
    }

    public Box CreateBox(string boxName, double price, double boxWidth,
        double boxLength, double boxHight, double boxThickness, string boxColor)
    {
        return _repository.CreateBox(boxName, price, boxWidth,
            boxLength, boxHight, boxThickness, boxColor);
    }

    public IEnumerable<Box> GetBoxes()
    {
        return _repository.GetBoxes();
    }

    public Box DeleteBox(int boxId)
    {
        return _repository.DeleteBox(boxId);
    }

    public Box GetBoxById(int boxId)
    {
        return _repository.GetBoxById(boxId);
    }

    public Box UpdateBox(Box updatedBox)
    {
        return _repository.UpdateBox(updatedBox);
    }
}

File: frontend/BoxFactoryFrontend/src/app/services/box.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import { Box, SearchBox } from '../models/box';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class BoxService {

  readonly url = 'http://localhost:5000/api'

  constructor(private http : HttpClient) {  }

  getBoxes(): Observable<Box[]> {
    return this.http.get<Box[]>(`${this.url}/boxes`);
  }

  createBox(request: Box): Observable<Box>{
    return this.http.post<Box>(`${this.url}/box`, request);
  }

  getBoxById(id: number): Observable<Box> {
    return this.http.get<Box>(`${this.url}/boxes/${id}`);
  }

  updateBox(id: number, box: Box): Observable<Box>{
    return this.http.put<Box>(`${this.url}/update/${id}`, box);
  }

  searchBoxes(criteria: string): Observable<Box[]> {
    return this.http.get<Box[]>(`${this.url}/search?searchTerm=${criteria}`);
  }

  deleteBox(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/delete/${id}`);
  }

}
